cpp/src/arrow/json/parser.cc

```diff
@@ -55,8 +55,8 @@ static Status ParseError(T&&... t) {
 }
 
 const std::string& Kind::Name(Kind::type kind) {
-  static const std::string names[] = {"null",   "boolean", "number",
-                                      "string", "array",   "object"};
+  static const std::string names[] = {"null",  "boolean", "number",          "string",
+                                      "array", "object",  "number_or_string"};
```

Comment:
The spacing is odd because it's trying to make a grid, so the start of `"string"` is aligned with the end of `"number_or_string"` on the next line.

FWIW, the preference I have acquired since writing this is to include a trailing comma in any array literal which will span multiple lines. `clang-format` recognizes this and puts each element on its own line, which looks much cleaner to me.

Reply:
Ah, makes sense. I added a comma after "number_or_string" and now clang-format puts all elements on the same line. Looks better now.